scripts/prepare_sector_network.py

```diff
@@ -3291,6 +3296,201 @@ def set_temporal_aggregation(n, opts, solver_name):
+
+    efficiency_orc = costs.at["organic rankine cycle", "efficiency"]
+    efficiency_dh = costs.at["geothermal", "efficiency residential heat"]
+
+    # p_nom_max conversion GW -> MW
+    egs_potentials["p_nom_max"] = egs_potentials["p_nom_max"] * 1000.0
+
+    # not using add_carrier_buses, as we are not interested in a Store
+    n.add("Carrier", "geothermal heat")
+
+    n.madd(
+        "Bus",
+        spatial.geothermal_heat.nodes,
+        location=spatial.geothermal_heat.locations,
+        carrier="geothermal heat",
+        unit="MWh_th",
+    )
+
+    n.madd(
+        "Generator",
+        spatial.geothermal_heat.nodes,
+        bus=spatial.geothermal_heat.nodes,
+        carrier="geothermal heat",
+        p_nom_extendable=True,
+    )
+
+    if snakemake.params.sector["enhanced_geothermal_var_cf"]:
+        efficiency = pd.read_csv(
+            snakemake.input.egs_capacity_factors, parse_dates=True, index_col=0
+        )
+        logger.info("Adding Enhanced Geothermal with time-varying capacity factors.")
+    else:
+        efficiency = pd.Series(1, overlap.index)
+
+    # if urban central heat exists, adds geothermal as CHP
+    as_chp = "urban central heat" in n.loads.carrier.unique()
+
+    if as_chp:
+        logger.info("Adding Enhanced Geothermal as Combined Heat and Power.")
+
+    for bus, bus_overlap in overlap.iterrows():
+        if not bus_overlap.sum():
+            continue
+
+        overlap = bus_overlap.loc[bus_overlap > 0.0]
+        bus_egs = egs_potentials.loc[overlap.index]
+
+        if not len(bus_egs):
+            continue
+
+        bus_egs["p_nom_max"] = bus_egs["p_nom_max"].multiply(bus_overlap)
+        bus_egs = bus_egs.loc[bus_egs.p_nom_max > 0.0]
+
+        if config["sector"]["enhanced_geothermal_performant"]:
+            bus_egs = bus_egs.sort_values(by="capital_cost").iloc[:1]
+            appendix = pd.Index([""])
+        else:
+            appendix = " " + pd.Index(np.arange(len(bus_egs)).astype(str))
+
+        # add surface bus
+        n.add(
+            "Bus",
+            f"{bus} geothermal heat surface",
+        )
+
+        bus_egs.index = np.arange(len(bus_egs)).astype(str)
+        well_name = f"{bus} enhanced geothermal" + appendix
+
+        bus_eta = pd.concat(
+            (efficiency[bus].rename(idx) for idx in well_name),
+            axis=1,
+        )
+
+        p_nom_max = bus_egs["p_nom_max"]
+        capital_cost = bus_egs["capital_cost"]
+        bus1 = pd.Series(f"{bus} geothermal heat surface", well_name)
+
+        n.madd(
+            "Link",
+            well_name,
+            location=bus,
+            bus0=spatial.geothermal_heat.nodes,
+            bus1=bus1,
+            carrier="geothermal heat",
+            p_nom_extendable=True,
+            p_nom_max=p_nom_max.set_axis(well_name) / efficiency_orc,
+            capital_cost=capital_cost.set_axis(well_name) * efficiency_orc,
+            efficiency=bus_eta,
```

Comment:
shouldn't the `capital_cost` depending on the `bus_eta` as well?

Reply:
Yup, good point, adjusted!